benchmark/sample_genes.py

```diff
@@ -74,6 +76,72 @@ def sample_genes(ortho_groups_fp,
     return genes
 
 
+def write_genes(genes,
+                input_faa_dir,
+                output_fa_dir,
+                output_genes_fp):
+    """ Write protein sequences of selected gene families to external files
+
+    Parameters
+    ----------
+    genes : dict of dict of set of str
+        { ogid : { taxon : set(protein(s)) }
+        return value of sample_genes or filter_paralogs
+    input_faa_dir : str
+        directory of input protein sequences from the query genome and the
+        selected taxa for comparison (FASTA format, one taxon per file)
+    output_fa_dir : str
+        directory to store output protein sequences (FASTA format, one gene
+        family per file)
+    output_genes_fp : str
+        file to store a list of selected gene families and their members.
+        format: gene1<tab>taxon1|protein1,taxon2|protein2,...
+    """
+    # generate a taxon to protein to gene family map
+    #   so complicated because it is optimized for subsequent filesystem I/O
+    prots = {}
+    for ogid in genes:
+        for taxon in genes[ogid]:
+            if taxon not in prots:
+                prots[taxon] = {}
+            for prot in genes[ogid][taxon]:
+                if prot in prots[taxon]:
+                    prots[taxon][prot].add(ogid)
+                else:
+                    prots[taxon][prot] = set([ogid])
```

Comment:
It is complicated! The logic is that: The FASTA files to be read (see a few line below), which are the reference proteome sequences, are big. It may be inefficient to load them all into memory. Therefore, it is important to sequentially read each line and check if this is the sequence I want, if yes then load it into memory. To do this test, the code needs two identifiers: taxon, and protein (taxon is the FASTA filename and protein is the sequence name). Therefore I translated the original data structure:
`{ gene : { taxon : { set of protein(s) } } }`
into:
`{ taxon : { protein : gene } }`
Do that sound reasonable?